esp/services/esdl_svc_engine/esdl_store.cpp

```diff
@@ -770,13 +795,43 @@ class CEsdlSDSStore : implements IEsdlStore, public CInterface
         esdldeftree->setProp("@id", lcId.str());
         esdldeftree->setProp("@esdlservice", esdlServiceName);
 
-        if (methodsConfig != nullptr)
-            esdldeftree->addPropTree("Methods", LINK(methodsConfig));
+#define IDENTIFY_ESDL_ENTRY(node, root, child) \
+	IPTree* node = nullptr; \
+    if (methodsConfig && (isEmptyString(root) || streq(methodsConfig->queryName(), root))) \
+    { \
+        node = methodsConfig; \
+        if (!isEmptyString(child)) \
+        { \
+            switch (node->getCount(child)) \
+            { \
+            case 0: \
+                methodsConfig = nullptr; \
+                break; \
+            case 1: \
+                methodsConfig = node->queryBranch(child); \
+                break; \
+            default: \
+                message.setf("Could not configure Service '%s' with multiple %s elements", esdlServiceName, child); \
+                return -1; \
+            } \
+        } \
+    }
+
+        IDENTIFY_ESDL_ENTRY(srcBinding, ESDL_BINDING_ENTRY, ESDL_DEF_ENTRY);
```

Comment:
IF we haven't ever allowed the caller to send in a definition section, and IF definition section should always a single service (unlike the ESDL interface definition part in DALI which eventually could have multiple services defined), do you think we could take this opportunity to have the user exposed name be Service instead of Definition.  Even if switched internally it might be more user friendly.

Reply:
The store has always assumed it was receiving Methods or nothing. But the calling service does accept Definition, and obviously returns Definition when retrieving a binding. I like the idea, but I think it should be its own JIRA that describes the end-to-end solution (acceptable input, what's saved, and what's returned).

Reply:
The problem with its own JIRA is just that we have to make sure this new feature is never released using one tag and then switched in a future release... or we have to forever maintain backward compatibility.

Reply:
if you think we already have to maintain bw compatibility then of course a new JIRA is fine.

Reply:
To summarize an offline discussion, the ESDL config service already processes binding input configurations that contain Binding and/or Definition elements. This story simply prevents those elements from being dropped by the service. Changing Definition to Service in the end user's command line interface does not eliminate the need to maintain support for Definition. Such changes are beyond the scope of this story and should be tracked by a new story.